Python/gc_free_threading.c

```diff
@@ -1245,6 +1491,19 @@ gc_collect_internal(PyInterpreterState *interp, struct collection_state *state,
 
     process_delayed_frees(interp, state);
 
+    #ifdef GC_ENABLE_MARK_ALIVE
+    if (!state->gcstate->freeze_active) {
```

Comment:
I think it's worth including your reasoning from the PR explaining the motivation for this condition as a code comment.